### PR TITLE
Publishing MCHTracking headers for Track and TrackParam

### PR DESCRIPTION
**Pull request by rpezzi**
@pillot , this is to use MCH tracking classes in root macros. 
* headers and src files split in separated directories.  
* o2_target_root_dictionary section for MCHTracking added to CMakeLists.txt

OK to proceed with this?

### CODE REVIEW

**Comment by rpezzi** [comment]
@pillot , @aphecetche , this is a lightweight dictionary with Track and TrackParam. Cluster.h moved to include/MCHTracking so that Track.h/cxx finds it. It may be possible to include with "../../src/Cluster.h" to avoid exposing Cluster.

File: Detectors/MUON/MCH/Tracking/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ o2_add_executable(trackfinderoriginal-workflow
         COMPONENT_NAME mch
         PUBLIC_LINK_LIBRARIES O2::MCHTracking)
 
-        o2_add_executable(trackfinder-workflow
+o2_add_executable(trackfinder-workflow
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Tracking/CMakeLists.txt:40]: as this file is finally unchanged, it should be removed from this PR

File: Detectors/MUON/MCH/Tracking/src/MCHTrackingLinkDef.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifdef __CLING__
+
+#pragma link C++ class o2::mch::Track + ;
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Tracking/src/MCHTrackingLinkDef.h:13]: is this LinkDef file needed in this end ?

Comment (rpezzi) [reply Detectors/MUON/MCH/Tracking/src/MCHTrackingLinkDef.h]: Useless, indeed.

**Comment by rpezzi** [comment]
While GPU Check seems unrelated to this PR, checkcode is failing unable to find Cluster.h and TrackParam.h at the new locations. Any idea?  

```
[ 28%] Building CXX object Detectors/MUON/MCH/Tracking/CMakeFiles/O2lib-MCHTracking.dir/src/Cluster.cxx.o
[ 28%] Built target O2exe-analysistutorial-new-collections
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/MUON/MCH/Tracking/src/Cluster.cxx:16:10: fatal error: MCHTracking/Cluster.h: No such file or directory
 #include "MCHTracking/Cluster.h"
          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [Detectors/MUON/MCH/Tracking/CMakeFiles/O2lib-MCHTracking.dir/src/Cluster.cxx.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 28%] Building CXX object Detectors/MUON/MCH/Tracking/CMakeFiles/O2lib-MCHTracking.dir/src/TrackParam.cxx.o
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/MUON/MCH/Tracking/src/TrackParam.cxx:16:10: fatal error: MCHTracking/TrackParam.h: No such file or directory
 #include "MCHTracking/TrackParam.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [Detectors/MUON/MCH/Tracking/CMakeFiles/O2lib-MCHTracking.dir/src/TrackParam.cxx.o] Error 1
gmake[1]: *** [Detectors/MUON/MCH/Tracking/CMakeFiles/O2lib-MCHTracking.dir/all] Error 2
```

**Comment by rpezzi** [comment]
@ktf , this PR just moves tree files to the include directory. Can you please check why o2checkcode is unable to find files at the new location?

**Pull request merged**